xarray/core/dataset.py

```diff
@@ -2318,20 +2337,87 @@ def rename(self, name_dict=None, inplace=None, **names):
         See Also
         --------
         Dataset.swap_dims
+        Dataset.rename_vars
+        Dataset.rename_dims
         DataArray.rename
         """
-        # TODO: add separate rename_vars and rename_dims methods.
-        inplace = _check_inplace(inplace)
         name_dict = either_dict_or_kwargs(name_dict, names, 'rename')
         for k, v in name_dict.items():
             if k not in self and k not in self.dims:
                 raise ValueError("cannot rename %r because it is not a "
                                  "variable or dimension in this dataset" % k)
 
         variables, coord_names, dims, indexes = self._rename_all(
-            name_dict=name_dict, dim_dict=name_dict)
+            name_dict=name_dict)
         return self._replace(variables, coord_names, dims=dims,
-                             indexes=indexes, inplace=inplace)
+                             indexes=indexes)
+
+    def rename_dims(self, name_dict=None, **names):
+        """Returns a new object with renamed dimensions only.
+
+        Parameters
+        ----------
+        name_dict : dict-like, optional
+            Dictionary whose keys are current dimension names and
+            whose values are the desired names.
+        **names, optional
+            Keyword form of ``name_dict``.
+            One of name_dict or names must be provided.
+
+        Returns
+        -------
+        renamed : Dataset
+            Dataset with renamed dimensions.
+
+        See Also
+        --------
+        Dataset.swap_dims
+        Dataset.rename
+        Dataset.rename_vars
+        DataArray.rename
+        """
+        name_dict = either_dict_or_kwargs(name_dict, names, 'rename')
+        for k, v in name_dict.items():
+            if k not in self.dims:
+                raise ValueError("cannot rename %r because it is not a "
+                                 "dimension in this dataset" % k)
+
+        dims = self._rename_dims(name_dict)
+        variables = self._rename_var_dims_only(name_dict)
+        indexes = self._rename_indexes(name_dict)
+        return self._replace(variables, dims=dims, indexes=indexes)
+
+    def rename_vars(self, name_dict=None, **names):
```

Comment:
Not to slow this down, but given it's a public method: anyone have thoughts on `rename_vars` vs `rename_variables`? I'm +0.2 on `rename_variables` for consistency with other methods.

Reply:
if we do rename_variables, should we also do rename_dimensions?

Reply:
I don't see `_variables` anywhere in https://xarray.pydata.org/en/stable/api.html#dataset. Seems like it's only used in relation to the advanced "Variable" API. Otherwise we do `data_vars`, `drop_vars` etc. It would also be consistent with abbreviating dimensions to `dims` and coordinates to `coords`.

Reply:
i.e. I'm in favor of `rename_vars`

Reply:
OK, let's do `rename_vars`. I was thinking the `.variables` property but abbreviations are good!